contrib/pgo-lto/Makefile

```diff
@@ -0,0 +1,65 @@
+.PHONY: top clean
+
+STAGE0_BUILD:=$(CURDIR)/stage0.build
+STAGE1_BUILD:=$(CURDIR)/stage1.build
+STAGE2_BUILD:=$(CURDIR)/stage2.build
+
+LLVM_BINDIR:=$(STAGE0_BUILD)/usr/tools/
+
+PROFILE_DIR:=$(CURDIR)/profiles
+PROFILE_FILE:=$(PROFILE_DIR)/merged.prof
+JULIA_ROOT:=$(CURDIR)/../..
+CXXFILT:=c++filt
+
+LLVM_PROFDATA:=$(LLVM_BINDIR)llvm-profdata
+export CC:=$(LLVM_BINDIR)clang
+export CXX:=$(LLVM_BINDIR)clang++
+export LD:=$(LLVM_BINDIR)ld.lld
+export AR:=$(LLVM_BINDIR)llvm-ar
+export RANLIB:=$(LLVM_BINDIR)llvm-ranlib
+
+# When building a single libLLVM.so we need to increase -vp-counters-per-site
+# significantly
+COUNTERS_PER_SITE:=6
+
+AFTER_STAGE1_MESSAGE:=Consider removing $(PROFILE_DIR) to start with a clean slate. $\
+    Then run Julia to collect realistic profile data, for example: \`julia -O3 -e $\
+    'using Pkg; Pkg.add("Unitful"); Pkg.test("Unitful")'\`. This should produce about $\
+	15MB of data in $(PROFILE_DIR). Note that running extensive scripts may result in $\
+	counter overflows, which can be detected by running \`make $(PROFILE_FILE)\`. $\
+	Afterwards run `make stage2`.
+
+$(STAGE0_BUILD) $(STAGE1_BUILD) $(STAGE2_BUILD):
+	$(MAKE) -C $(JULIA_ROOT) O=$@ configure
+
+stage0: USE_BINARYBUILDER_LLVM=1
+stage0: $(STAGE0_BUILD)
+	$(MAKE) -C $(STAGE0_BUILD)/deps install-clang install-llvm install-llvm-tools
+	# Turn [cd]tors into init/fini_array sections in libclang_rt, since lld
+	# doesn't do that, and otherwise the profile constructor is not executed
+	find $< -name 'libclang_rt.profile-*.a' -exec objcopy --rename-section .ctors=.init_array --rename-section .dtors=.fini_array {} +
```

Comment:
Do these not have opposite ordering?

Reply:
That could be, need to check. It's likely there's only one global so might not be an issue.

Reply:
```
$ nm --defined-only ./clang/14.0.5/lib/linux/libclang_rt.profile-x86_64.a | grep GLOBAL
0000000000000000 t _GLOBAL__sub_I_InstrProfilingRuntime.cpp
```